packages/block-library/src/image/save.js

```diff
@@ -27,18 +31,26 @@ export default function save( { attributes } ) {
 	} = attributes;
 
 	const newRel = isEmpty( rel ) ? undefined : rel;
+	const borderProps = getBorderClassesAndStyles( attributes );
 
 	const classes = classnames( {
 		[ `align${ align }` ]: align,
 		[ `size-${ sizeSlug }` ]: sizeSlug,
 		'is-resized': width || height,
+		'has-custom-border':
+			!! borderProps.className || ! isEmpty( borderProps.style ),
```

Comment:
This new `has-custom-border` class was required to be able to style image captions appropriately when in a Gallery block given border support now needs to be applied directly on the internal `img` element.